### ISSUE TITLE
All files are rebuilt on save when watching with "codedoc serve"

### ISSUE THREAD

**Issue by jamessimone** [CONNECT-platform/codedoc#16]
I have 24 pages (including _toc.md, 404.md, and index.md), which doesn't feel like a lot, but it takes some time (not a ton on my desktop, but quite a bit on my laptop) on sequential, individual, file saves to rebuild the entire directory.

It would be awesome if only the files being saved were rebuilt!

**Comment by loreanvictor** [comment]
Makes sense, I suspect it should be relatively easy to keep track of the last build time in memory and then check last change time of each markdown file using `fs.stat()`, ignoring those not changed after.

Just out of curiosity, roughly how long does it take to rebuild all of those 24 pages?

**Comment by jamessimone** [comment]
love `fs.stat()` - it currently takes ~17 seconds per save on my Thinkpad with 8 gigs of RAM and an Intel i5. 

It was really noticeable because I was going in to add the MetaOverride component into each file to fix the og:image issue we had been emailing about, but I have done these incremental changes a few times already in order to further "decorate" my files.

**Comment by loreanvictor** [comment]
wow thats a lot.

originally codedoc used to build all markdown files in parallel, which was pretty fast but also lead to memory issues (specifically on machines with lower RAM), so I had to change it. wonder if that should be a configurable thing (which of course would also require a better parallelization strategy than everything in parallel).
